### PR TITLE
menu-data: ES6ify. Tidying code.

### PR DESCRIPTION
**Pull request by retrofox**
Mainly this PR replaces functions by arrow functions which allows getting rid of use of bind.

### Testing

Run the application and then test the menus page for a site: `http://calypso.localhost:3000/menus/<your-testing-site>`

### CODE REVIEW

**Comment by retrofox** [comment]
/ cc @gwwar

File: client/lib/menu-data/menu-data.js [hunk 1]
```diff
@@ -388,29 +388,30 @@ MenuData.prototype.saveMenu = function( menu, callback ) {
...
-					this.emit( 'error', i18n.translate( 'There was a problem saving your menu.' ) );
-					debug( 'Error', error, data );
-					return;
-				}
+	wpcom
+	.undocumented()
+	.menusUpdate(
+		this.siteID,
+		menu.id,
+		this.interceptSaveForHomepageLink( menu ), ( error, data ) => {
```
Comment (gwwar) [inline client/lib/menu-data/menu-data.js]: Let's move the callback to the next line for scanability.

File: client/lib/menu-data/menu-data.js [hunk 2]
```diff
@@ -388,29 +388,30 @@ MenuData.prototype.saveMenu = function( menu, callback ) {
...
+			var parsedMenu;
```
Comment (gwwar) [inline client/lib/menu-data/menu-data.js]: I believe this can be const here.

Comment (retrofox) [inline client/lib/menu-data/menu-data.js]: yup. I should be defined below of the if condition.

**Comment by gwwar** [comment]
:+1: Verified that menus still work.

**Comment by retrofox** [comment]
:+1:

**Pull request merged**